modAionImpl/src/org/aion/zero/impl/sync/TaskImportBlocks.java

```diff
@@ -64,204 +80,622 @@
+     * value for the next LIGHTNING request. The returned base will be either retrieved from the set
+     * of previously generated values that have not yet been used or a new value generated by
+     * calling the given chain's {@link AionBlockchainImpl#nextBase(long, long)} method.
+     *
+     * @param best the starting point value for the next base
+     * @param baseSet list of already generated values
+     * @param chain the blockchain where the blocks will be imported which can be used to expand the
+     *     set of base value options
+     * @return the next base from the set or the given best value when the set does not contain any
+     *     values greater than it.
+     */
+    static long selectBase(
+            long best, long knownStatus, SortedSet<Long> baseSet, AionBlockchainImpl chain) {
+        // remove bases that are no longer relevant
+        while (!baseSet.isEmpty() && baseSet.first() <= best) {
+            baseSet.remove(baseSet.first());
         }
+
+        if (baseSet.isEmpty()) {
+            // generate new possible base value
+            return chain.nextBase(best, knownStatus);
+        } else {
+            Long first = baseSet.first();
+            baseSet.remove(first);
+            return first;
+        }
+    }
+
+    /**
+     * Utility method that determines if the given block is already stored in the given block store
+     * without going through the process of trying to import the block.
+     *
+     * @param store the block store that may contain the given block
+     * @param block the block for which we need to determine if it is already stored or not
+     * @return {@code true} if the given block exists in the block store, {@code false} otherwise.
+     * @apiNote Should be used when we aim to bypass any recovery methods set in place for importing
+     *     old blocks, for example when blocks are imported in {@link PeerState.Mode#FORWARD} mode.
+     */
+    static boolean isAlreadyStored(AionBlockStore store, AionBlock block) {
+        return store.getMaxNumber() >= block.getNumber() && store.isBlockExist(block.getHash());
     }
 
     private ImportResult importBlock(AionBlock b, String displayId, PeerState state) {
         ImportResult importResult;
         long t1 = System.currentTimeMillis();
         importResult = this.chain.tryToConnect(b);
         long t2 = System.currentTimeMillis();
-        log.info(
-                "<import-status: node = {}, sync mode = {}, hash = {}, number = {}, txs = {}, result = {}, time elapsed = {} ms>",
-                displayId,
-                (state != null ? state.getMode() : PeerState.Mode.NORMAL),
-                b.getShortHash(),
-                b.getNumber(),
-                b.getTransactionsList().size(),
-                importResult,
-                t2 - t1);
+        if (log.isDebugEnabled()) {
+            // printing sync mode only when debug is enabled
+            log.info(
```

Comment:
since it's a debugging log. should put it as the debug level log.

Reply:
updated in b5bbd67